apps/api/v2/src/modules/billing/billing.processor.ts

```diff
@@ -0,0 +1,74 @@
+import { BillingRepository } from "@/modules/billing/billing.repository";
+import { OrganizationsRepository } from "@/modules/organizations/organizations.repository";
+import { StripeService } from "@/modules/stripe/stripe.service";
+import { Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+
+export const INCREMENT_JOB = "increment";
+export const BILLING_QUEUE = "billing";
+export type IncrementJobDataType = {
+  oAuthClientId: string;
+  bookingId?: number | null;
+  startTime?: Date | null;
+  endTime?: Date | null;
+  userId?: number | null;
+};
+
+export type DecrementJobDataType = IncrementJobDataType;
+
+@Processor(BILLING_QUEUE)
+export class BillingProcessor {
```

Comment:
Question - how does `BillingProcessor` work? Is `handleIncrement` triggered once `BillingService.increaseUsageByClientId` adds `INCREMENT_JOB` to the `billingQueue` ?

Reply:
yes exactly